sdk/core/azure-core-management/src/main/java/com/azure/core/management/AzureEnvironment.java

```diff
@@ -133,6 +156,63 @@ public AzureEnvironment(Map<String, String> endpoints) {
             put("azureApplicationInsightsResourceId", "N/A");
         }});
 
+    /**
+     * Gets AzureEnvironment from Azure Resource Manager endpoint.
+     *
+     * If multiple Azure Resource Manager metadata is available, it uses the first one.
+     * Note that the Microsoft Graph endpoint currently is not available from metadata.
+     *
+     * @param endpoint the URL of Azure Resource Manager endpoint.
+     * @param httpPipeline the HTTP pipeline for requests.
+     * @return the AzureEnvironment
+     * @throws HttpResponseException thrown if the request is rejected by server, or failed to parse metadata.
+     * @throws IllegalArgumentException thrown if Azure Resource Manager endpoint is invalid.
+     * @throws NullPointerException thrown if Azure Resource Manager endpoint is null.
+     * @throws URISyntaxException thrown if Azure Resource Manager endpoint violates RFC 2396.
+     */
+    public static AzureEnvironment fromAzureResourceManagerEndpoint(String endpoint, HttpPipeline httpPipeline)
+        throws URISyntaxException {
```

Comment:
We generally don't throw checked exceptions in AzureSDK. Can we wrap URISyntaxException with `IllegalArgumentException`?